src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -149,6 +149,20 @@ def register_options(cls, register):
                 "We recommend keeping the default of `error` for CI builds."
             ),
         )
+        register(
+            "--run-against-entire-lockfile",
+            advanced=True,
+            default=False,
+            type=bool,
+            help=(
+                "If enabled, when running binaries, tests and repls Pants will use the entire "
```

Comment:
Nit, Oxford comma? Also I think you need a comma after the `when` clause

```suggestion
                "If enabled, when running binaries, tests, and repls, Pants will use the entire "
```